lib/src/commands/OptionsProcessor.test.ts

```diff
@@ -51,20 +51,243 @@ describe('navigation options', () => {
+                from: 0,
+                to: 1,
+                duration: 3,
+              },
+            },
+            dismissModal: {
+              enabled: true,
+              translationY: {
+                from: 0,
+                to: 1,
+                duration: 3,
+              },
+            },
+          },
+        };
+
+        const expected: Options = {
+          animations: {
+            showModal: {
+              enter: {
+                enabled: false,
+                translationY: {
+                  from: 0,
+                  to: 1,
+                  duration: 3,
+                },
+              },
+            },
+            dismissModal: {
+              exit: {
+                enabled: true,
+                translationY: {
+                  from: 0,
+                  to: 1,
+                  duration: 3,
+                },
+              },
+            },
+          },
+        };
+        uut.processOptions(options, CommandName.DismissModal);
+        expect(options).toEqual(expected);
+      });
+
+      it('processes old enabled options into new options,backwards compatibility ', () => {
+        const options: Options = {
+          animations: {
+            showModal: {
+              enabled: false,
+            },
+            dismissModal: {
+              enabled: true,
+            },
+          },
+        };
+
+        const expected: Options = {
+          animations: {
+            showModal: {
+              enter: {
+                enabled: false,
+              },
+            },
+            dismissModal: {
+              exit: {
+                enabled: true,
+              },
+            },
+          },
+        };
+        uut.processOptions(options, CommandName.DismissModal);
+        expect(options).toEqual(expected);
+      });
+
+      it('should not process new options', () => {
+        const options: Options = {
+          animations: {
+            showModal: {
+              enter: {
+                enabled: false,
+                translationY: {
+                  from: 0,
+                  to: 1,
+                  duration: 3,
+                },
+              },
+            },
+            dismissModal: {
+              exit: {
+                enabled: true,
+                translationY: {
+                  from: 0,
+                  to: 1,
+                  duration: 3,
+                },
+              },
+            },
+          },
+        };
+        const expected: Options = { ...options };
+        uut.processOptions(options, CommandName.DismissModal);
+        expect(options).toEqual(expected);
+      });
+    });
+  });
 
   it('keeps original values if values were not processed', () => {
     const options: Options = {
       blurOnUnmount: false,
       popGesture: false,
       modalPresentationStyle: OptionsModalPresentationStyle.fullScreen,
-      animations: { dismissModal: { alpha: { from: 0, to: 1 } } },
```

Comment:
I added a whole suite about it `Modal Animation Options` that covers this case and other cases, removed for concern separation of the test since we have dedicated suite about it.

Reply:
it an be refactored with the comment below, but the case moved to another place `processes old options into new options,backwards compatibility`